### ISSUE TITLE
RTX 3000 series multi GPU slower than on 1 gpu

### ISSUE THREAD

**Issue by JulianPinzaru** [NVlabs/stylegan2-ada-pytorch#68]
Hello !
I've got a problem running stylegan2-ada-pytorch on 2 gpus (RTX 3090 FE). 
It appears that it's not working well in multiprocessing environment and I am not sure what would be the reason for that.  
So I have tried different combinations of latest cuda and pytorch. It always works well when training on 1 GPU though and it doesn't when trying on 2 GPUs (while having exactly same settings / environment).

My specs:
- **Nvidia driver**: 460.39 
- **CUDA version**:  11.1.1
- **GPUs**: 2 RTX 3090 FE
- **Pytorch**: 1.8.0
- **Stylegan2-ada-pytorch nvlabs repo**: yes (original)

The command I run:
`python train.py --outdir=./results/ --data=./datasets/jjl_512/ --gpus=2 --snap=4 --metrics=None --mirror=1 --cfg=auto --nhwc=True --aug=fixed --p=0.250 --augpipe=geom --resume=ffhq512 --workers=8`

So it works fine with that configuration on 1 GPU, but it is slower on 2 GPUs.
![image](https://user-images.githubusercontent.com/51966133/111130018-fa0e4e80-8533-11eb-804d-1c6d3da87941.png)

I would be very thankful if there is any suggestions on how to fix that problem for me (and other people perhaps).
Thank you in advance!

**Comment by entmike** [comment]
I see that you closed this issue.  Did you resolve it?  I am struggling to get the right parameters to get better than 1.5x performance improvement on a 1024x1024 dataset.

**Comment by JulianPinzaru** [comment]
> I see that you closed this issue. Did you resolve it? I am struggling to get the right parameters to get better than 1.5x performance improvement on a 1024x1024 dataset.

Hi! I dont think it's related, it was a problem in the code for me because I changed some values in config which I shouldnt have done.
You must be running well with cfg auto, --gpus=2 , I guess. 
For me it runs cfg auto (512x512 dataset) 4min30sec on 1 gpu and 2min30sec on 2 gpus. I tweaked it later to save 20 more seconds (increased batch sizes to fill up the vram)
